### ISSUE TITLE
Error Getting Named Credentials: Timeout trying to establish connection to iframe

### ISSUE THREAD

**Issue by robertwynter** [douglascayers-org/sfdx-mass-action-scheduler#123]
Any time I go into MA and select a configuration for one of my processes it errors
![Mass action error](https://user-images.githubusercontent.com/6217191/64922384-ad11b680-d79c-11e9-9562-3f9ce08b4555.PNG)
 and times out

**Comment by Clint-Chester** [comment]
Unfortunately getting the following error thrown quite intermittently (pending on the environment) to help with the debugging of this:

```
exceptionType: "System.VisualforceException"
isUserDefinedException: false
message: "org.apache.commons.httpclient.RedirectException: Maximum redirects (100) exceeded"
stackTrace: "Class.dca_mass_action.LC_VisualforceDomainController.getVisualforceDomainURL: line 14, column 1"
```

**Comment by robertwynter** [comment]
sorry, I'm not sure what this means. Would you mind expanding on what you are explaining?

**Comment by kritikaraj303** [comment]
Facing the similiar issue, tried 
1.Allow IFraming of Visualforce Pages with Clickjack Protection , 

2.Add the same Visualforce domains for clickjack protection to the Security | CORS menu in Setup according to Use CORS to Access Salesforce Resources from Web Browsers. ,

3.Ensure the users encountering the issue have been assigned access to the Visualforce pages shipped with Mass Action Scheduler.
LC_APIPage
LC_VisualforceDomainPage

But still not working. Please , look into it as it is very urgent.

**Comment by efmercado** [comment]
Has anyone been able to resolve this issue? I recently installed this package and am too unable to access it. The error. attached below, appears when I try to create a new record.

<img width="1440" alt="LPI Error Message" src="https://user-images.githubusercontent.com/56619896/68059539-15b7e280-fcba-11e9-9156-085313fe0e78.png">

**Comment by robertwynter** [comment]
Mine just simply started working on it’s own. So I’m not sure what the issue was on our end.

**Comment by douglascayers** [comment]
Unfortunately I have no idea what causes the network hiccup between the Lightning Components sometimes able and sometimes not able to reach the Visualforce domain.

Thanks @robertwynter for confirming your issue eventually worked out.

Everyone else, some things people try with varying degrees of success:
* Log out and log back in
* Refresh the page a few times
* Wait a few minutes and try again

**Comment by douglascayers** [comment]
I believe I resolved this in the underlying library I use to make API calls based on https://github.com/douglascayers/sfdx-lightning-api-component/pull/7.

I'll need to merge that change into this project to fix it for Mass Action Scheduler. I have high expectations that change and https://github.com/douglascayers-org/sfdx-mass-action-scheduler/pull/115/commits/6dec5c35079d67e82db09836f2b068143b3c2efc will resolve the inconsistent "cannot connect" errors.

**Comment by douglascayers** [comment]
Fixed per #115

**Comment by gcoogan-athene** [comment]
@douglascayers,
These exact timeout errors are now occurring after upgrading from 2.302 to 2.4 in a sandbox.
I attempted to remove the connected app, named credential, and auth provider, then recreate them in configuration screen. This did not resolve the issue. Should I open a new issue?
